pulp_container/app/registry_api.py

```diff
@@ -345,36 +349,132 @@ def get(self, request):
         return Response(data={})
 
 
-class CatalogView(ContainerRegistryApiMixin, APIView):
+class ContainerCatalogSerializer(ModelSerializer):
+    """
+    Serializer for Distributions in the _catalog endpoint of the registry.
+    """
+
+    class Meta:
+        model = models.ContainerDistribution
+        fields = ["base_path"]
+
+
+class ContainerCatalogPagination(BasePagination):
+    """
+    Pagination class to paginate repositories by names according to the registry api specification.
+    """
+
+    def paginate_queryset(self, queryset, request, view=None):
+        """
+        Analyse the pagination parameters and prepare the queryset.
+        """
+        try:
+            self.n = int(request.query_params.get("n"))
+        except Exception:
+            self.n = self.page_size
+        last = request.query_params.get("last")
+        self.url = request.build_absolute_uri()
+
+        if last:
+            queryset = queryset.filter(base_path__gt=last)
+        return queryset.order_by("base_path")[: self.n]
+
+    def get_paginated_response(self, data):
+        """
+        Prepare the paginated container _catalog response.
+        """
+        repositories_names = [repo["base_path"] for repo in data]
+        parsed_url = list(urlparse(self.url))
+        query_params = parse_qs(parsed_url[4])
+        query_params["n"] = str(self.n)
+        query_params["last"] = repositories_names[-1]
```

Comment:
this will give en error if repo_names is empty list

Reply:
That needs fixing for sure!